src/Castle.Core.Tests/DynamicProxy.Tests/ComInterfacesTests.cs

```diff
@@ -29,29 +29,53 @@ public class ComInterfacesTests:BasePEVerifyTestCase
 		[Test]
 		public void Marshal_Release_throws_when_called_with_IntPtr_Zero()
 		{
-			Assert.Throws<ArgumentNullException>(() => Marshal.Release(IntPtr.Zero));
+			Assert.Catch<ArgumentException>(() => Marshal.Release(IntPtr.Zero));
+			//           ^^^^^^^^^^^^^^^^^
+			// We're not going to be more precise because Mono and the CLR don't throw the same exception type.
+			// See https://github.com/mono/mono/issues/15853.
 		}
 
+		// NOTE: You may wonder why the following tests delegate their implementation to static methods
+		// in a private static class. This is because the method bodies contain references to ADODB.dll,
+		// which fails to load under Mono/Linux, causing test discovery to abort early. By moving these
+		// method bodies somewhere else, test discovery will succeed and we get a chance at specifying
+		// how the tests should run, e.g. using the `[Platform]` attributes.
+
 		[Test]
+		[Platform(Include = "Win", Reason = "Depends on ADODB.dll, which can fail to load under Mono/Linux.")]
 		public void Can_proxy_ADO_RecorsSet()
 		{
-			generator.CreateInterfaceProxyWithoutTarget<Recordset>();
+			Implementation.Can_proxy_ADO_RecorsSet(generator);
 		}
 
 		[Test]
+		[Platform(Include = "Win", Reason = "Depends on OLE32.dll due to co-class instantiation, which is likely only available on Windows.")]
 		public void Can_proxy_existing_com_object()
 		{
-			var command = new Command();
-			var parameter = ExtractActualComParameterObject(command.CreateParameter("foo"));
-			generator.CreateInterfaceProxyWithTargetInterface(parameter);
+			Implementation.Can_proxy_existing_com_object(generator);
 		}
 
-		private Parameter ExtractActualComParameterObject(Parameter parameter)
+		private static class Implementation
```

Comment:
Just so I've got this right, this is relying on the JIT not loading the private class during test discovery?

Another option would be to lazily reference/load adodb with reflection, and obviously call `CreateInterfaceProxyWithoutTarget` with a type variable and use reflection to new the `Command` class.

I'd prefer it to be correct code rather than relying on a JIT behaviour.

Reply:
> I'd prefer it to be correct code rather than relying on a JIT behaviour.

I was just thinking that we could completely eliminate JIT / assembly loader / version binding policy failure as an error source if we simply manually embedded the needed COM interop type definitions in our test project. We can open up `ADODB.dll` in ILDASM or ILSpy, then bring the required type definitions over into our own project. Then we could remove `ADODB.dll` from the solution.

Reply:
> manually embedded the needed COM interop type definitions in our test project

👍. That one isn't a PIA so there shouldn't be any reliance on being signed a certain way.

Reply:
OK, will do this then. It'll be nice to say bye to that DLL. :smirk:

Reply:
Done. I've had to make a somewhat arbitrary choice about which ADO types to bring in fully, and which to reproduce only partially (and where to make the cut in the latter cases). I've brought in all required enums, but truncated co-class types as well as the interface hierarchies. (This shouldn't matter since the vtables of derived COM interfaces are a superset of the base interfaces' vtables, and because we're only calling methods on interfaces, not on any co-classes directly.) I've also opted to omit custom marshalers, and COM event interfaces. I've however chosen to reproduce method signatures as accurately as possible.